Libplanet/Net/Protocols/RoutingTable.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Nito.AsyncEx;
+using Serilog;
+
+namespace Libplanet.Net.Protocols
+{
+    internal class RoutingTable
+    {
+        private readonly Address _address;
+        private readonly int _tableSize;
+        private readonly int _bucketSize;
+        private readonly Random _random;
+        private readonly KBucket[] _buckets;
+        private readonly AsyncLock _bucketMutex;
+
+        public RoutingTable(
```

Comment:
It would be great if we have unit tests for this class.

Reply:
I wanted to add tests using distance, but generating `PublicKey` with specified `Address` was quite picky. I will work on more precise tests later.